lib/pbench/test/functional/agent/cli/commands/tools/test_tools_list.py

```diff
@@ -259,24 +259,37 @@ def tools_on_multiple_hosts(self, pbench_run):
                     tool = p / "perf"
                     tool.write_text("--record-opts='-a --freq=100'")
 
-    # Issue #3434
+    # Issue #3454
     @pytest.fixture
     def labels_on_multiple_hosts(self, pbench_run):
+        # This fixture is meant to be called after the previous one
+        # (tools_on_multiple_hosts). The previous one establishes a
+        # tool-like directory structure; this one just embellishes it
```

Comment:
Consider making this fixture depend on the previous one, so that the dependency is explicit and cannot be gotten wrong.
```python
    @pytest.fixture
    def labels_on_multiple_hosts(self, pbench_run, tools_on_multiple_hosts):
```

Reply:
Yes, much cleaner - thanks!